sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthenticationUtil.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.BearerTokenAuthenticationPolicy;
+import com.azure.core.util.Context;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Supplier;
+
+public class AuthenticationUtil {
```

Comment:
Make this class final and add a private constructor (to prevent instantiation of this class).